backend/src/main/java/org/sefglobal/core/academix/controller/admin/SubCategoryAdminController.java

```diff
@@ -24,12 +24,11 @@ public SubCategoryAdminController(SubCategoryService subCategoryService) {
         this.subCategoryService = subCategoryService;
     }
 
-    @PostMapping("/{categoryId}")
+    @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public SubCategory addSubCategory(@PathVariable long categoryId,
-                                      @Valid @RequestBody SubCategory subCategory)
+    public SubCategory addSubCategory(@Valid @RequestBody SubCategory subCategory)
             throws ResourceNotFoundException {
-        return subCategoryService.addSubCategory(categoryId, subCategory);
+        return subCategoryService.addSubCategory(subCategory.getCategory().getId(), subCategory);
```

Comment:
Can't we just send the object itself to the addSubCategory method on service? Is it necessary to send the Id separately?

Reply:
Ah right. I'll update addSubCategory service

Reply:
Resolved